crypto/aes/asm/aesv8-armx.pl

```diff
@@ -211,7 +211,12 @@
 .Loop192:
 	vtbl.8	$key,{$in1},$mask
 	vext.8	$tmp,$zero,$in0,#12
+#ifdef __ARMEB__
+	vst1.32	{$in1},[$out],#16
+	sub	$out,$out,#8
```

Comment:
You can use vstl instruction like 'vst1 {v4.d}[1], [$out],#8' to get the higher 64bits stored so that you need not have to do the following sub instruction.

Reply:
I agree your idea,that SIMD vector element names can get higher 64bits,directly,thanks!

Reply:
first,I test your suggest as this ,get the result as below:( vst1 {v4.d}[0], [$out],#8)
-----------------------------------------------------------------------------------------------------------
Breakpoint 3, aes_v8_set_encrypt_key () at aesv8-armx2.S:105
105		ext	v5.16b,v0.16b,v3.16b,#12
(gdb) info registers x2
x2             0xfffefa28	4294900264
(gdb) x/8wx 0xfffefa28
0xfffefa28:	0x00000000	0x00000000	0x00000000	0x00000000
0xfffefa38:	0x00000000	0x00000000	0x00000000	0x00000000
(gdb) info registers v4
v4             {d = {f = {0x0, 0x0}, u = {0x0, 0x3fa3199bb202435a}, s = {0x0, 0x3fa3199bb202435a}}, s = {
    f = {0x0, 0x0, 0x1, 0x0}, u = {0x0, 0x0, 0x3fa3199b, 0xb202435a}, s = {0x0, 0x0, 0x3fa3199b,
      0xb202435a}}, h = {u = {0x0, 0x0, 0x0, 0x0, 0x3fa3, 0x199b, 0xb202, 0x435a}, s = {0x0, 0x0, 0x0,
      0x0, 0x3fa3, 0x199b, 0xb202, 0x435a}}, b = {u = {0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3f,
      0xa3, 0x19, 0x9b, 0xb2, 0x2, 0x43, 0x5a}, s = {0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3f, 0xa3,
      0x19, 0x9b, 0xb2, 0x2, 0x43, 0x5a}}, q = {u = {0x00000000000000003fa3199bb202435a}, s = {
      0x00000000000000003fa3199bb202435a}}}
(gdb) n
107		st1	{v4.d}[0], [x2],#8
(gdb) n
111		aese	v6.16b,v0.16b
(gdb) x/8wx 0xfffefa28
0xfffefa28:	0x3fa3199b	0xb202435a	0x00000000	0x00000000
0xfffefa38:	0x00000000	0x00000000	0x00000000	0x00000000
--------------------------------------------------------------------------------------------------------
as i modified ,the result is :  
(gdb) info registers x2
x2             0xfffefa28	4294900264
(gdb) x/8wx  0xfffefa28
0xfffefa28:	0x00000000	0x00000000	0x00000000	0x00000000
0xfffefa38:	0x00000000	0x00000000	0x00000000	0x00000000
(gdb) n
107		st1	{v4.4s},[x2],#16
(gdb) x/8wx  0xfffefa28
0xfffefa28:	0x00000000	0x00000000	0x00000000	0x00000000
0xfffefa38:	0x00000000	0x00000000	0x00000000	0x00000000
(gdb) n
108		sub	x2,x2,#8
(gdb) x/8wx  0xfffefa28
0xfffefa28:	0xb202435a	0x3fa3199b	0x00000000	0x00000000
0xfffefa38:	0x00000000	0x00000000	0x00000000	0x00000000
(gdb) info registers v4
v4             {d = {f = {0x0, 0x0}, u = {0x0, 0x3fa3199bb202435a}, s = {0x0, 0x3fa3199bb202435a}}, s = {f = {0x0, 0x0,
      0x1, 0x0}, u = {0x0, 0x0, 0x3fa3199b, 0xb202435a}, s = {0x0, 0x0, 0x3fa3199b, 0xb202435a}}, h = {u = {0x0, 0x0, 0x0,
      0x0, 0x3fa3, 0x199b, 0xb202, 0x435a}, s = {0x0, 0x0, 0x0, 0x0, 0x3fa3, 0x199b, 0xb202, 0x435a}}, b = {u = {0x0, 0x0,
      0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3f, 0xa3, 0x19, 0x9b, 0xb2, 0x2, 0x43, 0x5a}, s = {0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
      0x0, 0x0, 0x3f, 0xa3, 0x19, 0x9b, 0xb2, 0x2, 0x43, 0x5a}}, q = {u = {0x00000000000000003fa3199bb202435a}, s = {
      0x00000000000000003fa3199bb202435a}}}

----------------------------------------------------------------------------------------------------
0xfffefa28:	0x3fa3199b	0xb202435a

0xfffefa28:	0xb202435a	0x3fa3199b

there are different for the order,so I can't take your opinion.

Reply:
I'm a bit confused. In your original implementation, the value you want to store is 0x00000000000000003fa3199bb202435a (in v4 register), after 'st1 {v4.4s},[x2],#16', the value at address 0xfffefa28 is  '0xfffefa28: 0xb202435a 0x3fa3199b 0x00000000 0x00000000'.  It looks like the behavior of Little-Endian rather than Big-Endian.  Shouldn't the value at address 0xfffefa28  be '0xfffefa28: 0x00000000 0x00000000 0x3fa3199b 0xb202435a' for Big-Endian?

Reply:
It is a Big-Endian,
(gdb) set  *(int*)0xfffefa30=0x01020304
(gdb) x/16bx 0xfffefa30
0xfffefa30:	0x01	0x02	0x03	0x04	0x05	0x06	0x07	0x08

++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
s = {0x43a2313f7c0128a4ce036bfef1a07265}}}
(gdb) n
107		st1	{v4.4s},[x2],#16
(gdb) x/16bx 0xfffefab0
0xfffefab0:	0xf1	0xa0	0x72	0x65	0xce	0x03	0x6b	0xfe
0xfffefab8:	0x7c	0x01	0x28	0xa4	0x43	0xa2	0x31	0x3f
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
s = {0x43a2313f7c0128a4ce036bfef1a07265}}}
(gdb) n
107		st1	{v4.d}[0], [x2],#8
(gdb) n
111		aese	v6.16b,v0.16b
(gdb) x/16bx 0xfffefa30
0xfffefa30:	0xce	0x03	0x6b	0xfe	0xf1	0xa0	0x72	0x65
0xfffefa38:	0x01	0x02	0x03	0x04	0x05	0x06	0x07	0x08

Reply:
Good, thanks for your explanation.